clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/metrics/UrlMetricsInstrumentation.groovy

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.docker.registry.metrics
+
+import com.netflix.spectator.api.Id
+import com.netflix.spectator.api.Registry
+import com.squareup.okhttp.Response
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import java.util.concurrent.TimeUnit
+
+import static java.lang.String.format
+
+@Slf4j
+@Component
+class UrlMetricsInstrumentation {
+
+  private final Registry registry
+
+  private final Id timingId
+  private final Id counterId
+
+  @Autowired
+  UrlMetricsInstrumentation(Registry registry) {
+    this.registry = registry
+    timingId = registry.createId('UrlRequestTime').withTag('className', UrlMetricsInstrumentation.simpleName)
+    counterId = registry.createId('UrlCount').withTag('className', UrlMetricsInstrumentation.simpleName)
+  }
+
+  void interceptorCompleted(Response response, long start, long end) {
+    registry.timer(timingId.withTag('url', response.request().url().toString())).record(end - start, TimeUnit.MILLISECONDS)
+    String status = (response.successful) ? 'success' : 'failure'
+    registry.counter(counterId.withTag('url', response.request().url().toString()).withTag('status', status)).increment()
+    log.info format("Received response for %s in %.1fms", response.request().url(), (end - start) / 1e6d)
```

Comment:
Personally for me `1_000_000.0` is more readable than `1e6d`

Reply:
Which is the minimal supported Java version for Spinnaker/Clouddriver? If 7, then `1_000_000.0` is superb, if older than 7, than it's not supported

Reply:
Better yet, turn `1e6d` or `1_000_000.0` into a static variable with an useful name, so we actually know what this magic number means

Reply:
Better yet, use the JDK time methods to do the transform so that it's clear what you're doing: https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html

Reply:
Updated, using java.time. Start and End are measured using `Instant`, and then a `Duration` is calculated between them which is then converted to milliseconds. No more `1e6d` magic!